web/permissions/permissions.go

```diff
@@ -119,7 +120,8 @@ func extract(c echo.Context) (*permissions.Claims, *permissions.Set, error) {
 	} else if claims.Audience == permissions.AccessTokenAudience {
 		// Oauth token, extract permissions from JWT-encoded scope
 		pset, err = permissions.UnmarshalScopeString(claims.Scope)
-	} else if claims.Audience == permissions.RegistrationTokenAudience {
+	} else if registerToken := []byte(c.QueryParam("registerToken"));
+			subtle.ConstantTimeCompare(registerToken, instance.RegisterToken) == 1 {
```

Comment:
2 things:

1. This test is too late. The registerToken is used when there is no JWT token, and so, we have returned at line 99.
2. We should do the constant time comparison only when there is a token. No need to burn CPU cycles when there is no token at all.